qiime2/core/cache.py

```diff
@@ -191,6 +215,18 @@ def __init__(self, path=None, process_pool_lifespan=45):
         # We were used by this process
         USED_CACHES.add(self)
 
+        # Start thread that pokes things in the cache to ensre they aren't
+        # culled for being too old
+        self._thread_is_done = threading.Event()
```

Comment:
@Oddant1, I realized on the walk home that we really only want to do this if `path` is `None`, since we should expect user-provided caches to be stable to this sort of nonsense.